src/java/org/apache/cassandra/schema/SchemaConstants.java

```diff
@@ -77,14 +78,32 @@ public final class SchemaConstants
      */
     public static final int NAME_LENGTH = 48;
 
+    /**
+     * Longest acceptable file name. Longer names lead to too long file name error.
+     */
+    public static final int FILENAME_LENGTH = 255;
+
+    /**
+     * Longest acceptable table name, so it can be used in a directory
+     * name constructed with a suffix of a table id and a separator.
+     */
+    public static final int TABLE_NAME_LENGTH = FILENAME_LENGTH - 32 - 1;
```

Comment:
what is about introducing an extra constant here instead of a magic number (32), like: 
TABLE_UUID_AS_HEX_LENGTH = 32
to make the logic a bit more clear?